runtime/src/iree/hal/drivers/cuda/cuda_allocator.c

```diff
@@ -236,12 +236,12 @@ iree_hal_cuda_allocator_query_buffer_compatibility(
   if (iree_all_bits_set(params->type, IREE_HAL_MEMORY_TYPE_DEVICE_VISIBLE)) {
     compatibility |= IREE_HAL_BUFFER_COMPATIBILITY_IMPORTABLE;
   }
+  if (iree_any_bit_set(params->usage, IREE_HAL_BUFFER_USAGE_TRANSFER)) {
```

Comment:
I don't think this should be required - did you hit an error?

(if we make changes here we also have to do cuda2/cuda_allocator.c)

Reply:
ah maybe with staging buffers - we should make sure cuda2 (and rocm) get this fix

Reply:
IIRC some buffers are `DEVICE_LOCAL` but not `DEVICE_VISIBLE`, so the `QUEUE_TRANSFER` won't be set and hence we can't map some buffers sometimes. 

Let me rebase this on top of main real quick and re-discover the bug real quick. :)
FWIW, we have been running this on SRT for our Llama on python speedup purposes.

Reply:
all DEVICE_LOCAL buffers are DEVICE_VISIBLE - but if it's a HOST_LOCAL buffer it may not be DEVICE_VISIBLE
thx for rebasing - we can double check before changing the semantics

Reply:
I think that might be it. but will confirm in a bit, I need to rebuild IREE on my local machine. :D

Reply:
@benvanik Seems like the issue is not there anymore, I used to need it to pass the `array_interop_test.py` with cuda backend. I will remove the changes in allocator now.